pkg/database/sync.go

```diff
@@ -57,24 +58,27 @@ func (dao *DAO) SyncData(ctx context.Context, event model.SyncEvent,
 
 		// TODO: Need better safety for delete errors.
 		// The current retry logic won't work well if there's an error here.
-		batch.Queue(batchItem{
+		err := batch.Queue(batchItem{
```

Comment:
Why don't we use queueErr variable here?

Reply:
It causes a lint error because the value gets updated right after.
Ideally I should check after each and return, but I'm trying to avoid adding too much complexity.